Test failure when using PYCA_WINDOWS_LINK_TYPE=dynamic

Running the tests fails when using PYCA_WINDOWS_LINK_TYPE=dynamic to get dynamic linking.

Test was run via:

set PYCA_WINDOWS_LINK_TYPE=dynamic
python setup.py test

```
============================= test session starts =============================
platform win32 -- Python 2.7.9 -- py-1.4.26 -- pytest-2.6.4
collected 74761 items
```

The test should not check for ssleay32mt in that case.

```
================================== FAILURES ===================================
_________________________ TestOpenSSL.test_libraries __________________________

self = <test_openssl.TestOpenSSL object at 0x1C1E3A10>

    def test_libraries(self):
        assert _get_libraries("darwin") == ["ssl", "crypto"]
>       assert "ssleay32mt" in _get_libraries("win32")
E       assert 'ssleay32mt' in ['libeay32', 'ssleay32', 'advapi32']
E        +  where ['libeay32', 'ssleay32', 'advapi32'] = _get_libraries('win32')

tests\hazmat\bindings\test_openssl.py:139: AssertionError
```

--- a/tests/hazmat/bindings/test_openssl.py
+++ b/tests/hazmat/bindings/test_openssl.py
@@ -4,8 +4,6 @@
 
 from __future__ import absolute_import, division, print_function
 
-import os
-
 import pytest
 
 from cryptography.hazmat.bindings.openssl.binding import (
@@ -138,7 +136,7 @@ class TestOpenSSL(object):
 
     def test_libraries(self, monkeypatch):
         assert _get_libraries("darwin") == ["ssl", "crypto"]
-        monkeypatch.setitem(os.environ, 'PYCA_WINDOWS_LINK_TYPE', 'static')
+        monkeypatch.setenv('PYCA_WINDOWS_LINK_TYPE', 'static')
         assert "ssleay32mt" in _get_libraries("win32")
 
     def test_windows_static_dynamic_libraries(self):

--- a/tests/hazmat/bindings/test_openssl.py
+++ b/tests/hazmat/bindings/test_openssl.py
@@ -4,6 +4,8 @@
 
 from __future__ import absolute_import, division, print_function
 
+import os
+
 import pytest
 
 from cryptography.hazmat.bindings.openssl.binding import (
@@ -134,8 +136,9 @@ class TestOpenSSL(object):
         assert resp == expected_options
         assert b.lib.SSL_get_mode(ssl) == expected_options
 
-    def test_libraries(self):
+    def test_libraries(self, monkeypatch):
         assert _get_libraries("darwin") == ["ssl", "crypto"]
+        monkeypatch.setitem(os.environ, 'PYCA_WINDOWS_LINK_TYPE', 'static')
         assert "ssleay32mt" in _get_libraries("win32")
 
     def test_windows_static_dynamic_libraries(self):

